### PR TITLE
refactor(insights): Improve clarity of async querying and timeout handling

### PR DESCRIPTION
**Pull request by Twixes**
## Changes

Finally dove into how the async queries system works and cleaned a few small things up along the way. The only behavioral change is that we'll now return HTTP code 202 for async queries, which is more correct semantically: it means "the request has been accepted but you'll have to check back for results", differently from 200 which means "OK, here are the results".

## How did you test this code?

No behavior changes, except for the 202 one – I couldn't find an existing test for async query requests though, let me know where to put one in so that the 202 behavior is codified.

### CODE REVIEW

**Comment by webjunkie** [comment]
There are tests in `posthog/api/test/test_query.py`, however, you are right that they don't test making an actual async query as far as I can see. Mostly that retrieving (with mocks) works.

File: frontend/src/queries/query.ts [hunk 1]
```diff
@@ -93,6 +93,8 @@ export function queryExportContext<N extends DataNode = DataNode>(
     }
 }
 
+const SYNC_ONLY_QUERY_KINDS = ['HogQLMetadata', 'EventsQuery', 'HogQLAutocomplete'] satisfies NodeKind[keyof NodeKind][]
```
Comment (mariusandra) [inline frontend/src/queries/query.ts:96]: Why `EventsQuery` here?

Comment (Twixes) [reply frontend/src/queries/query.ts:96]: That's a question for @webjunkie I believe

Comment (webjunkie) [reply frontend/src/queries/query.ts:96]: It was.... you @mariusandra 👀 

https://github.com/PostHog/posthog/pull/18907

**Comment by Twixes** [comment]
Thanks @webjunkie I see now. Added a test, take a look!

**Review by webjunkie — APPROVED** [review]

**Pull request merged**